test/javascripts/acceptance/core-sidebar-test.js

```diff
@@ -0,0 +1,195 @@
+            chatable: { slug: "general" },
+          },
+          {
+            id: 3,
+            title: "random",
+            unread_count: 1,
+            unread_mentions: 1,
+            chatable_type: "Category",
+            chatable: { slug: "random" },
+          },
+        ],
+        direct_message_channels: directChannels,
+      });
+    });
+  });
+
+  test("Public channels section", async function (assert) {
+    await visit("/");
+
+    assert.strictEqual(
+      query(
+        ".sidebar-section-chat-channels .sidebar-section-header-text"
+      ).textContent.trim(),
+      "Channels",
+      "displays correct channels section title"
+    );
+    const links = queryAll(".sidebar-section-chat-channels a");
+    assert.strictEqual(
+      links
+        .eq(0)
+        .find(".sidebar-section-link-prefix svg")[0]
+        .classList.contains("d-icon-hashtag"),
+      true,
+      "displays hash icon prefix"
+    );
+    assert.strictEqual(
+      links
+        .eq(0)
+        .find(".sidebar-section-link-prefix svg.prefix-badge")[0]
+        .classList.contains("d-icon-lock"),
+      true,
+      "displays lock badge for restricted channel"
+    );
+    assert.strictEqual(
+      links[0].textContent.trim(),
+      "dev",
+      "displays dev name in a link"
+    );
+    assert.ok(
+      !exists(links.eq(0).find(".sidebar-section-link-suffix")[0]),
+      "does not display new messages indicator"
+    );
+
+    assert.strictEqual(
+      links
+        .eq(1)
+        .find(".sidebar-section-link-prefix svg")[0]
+        .classList.contains("d-icon-hashtag"),
+      true,
+      "displays hash icon prefix"
+    );
+    assert.ok(
+      !exists(
+        links.eq(1).find(".sidebar-section-link-prefix svg.prefix-badge")[0]
+      ),
+      "does not display lock badge for public channels"
+    );
+    assert.strictEqual(
+      links[1].textContent.trim(),
+      "general",
+      "displays general name in a link"
+    );
+    assert.strictEqual(
+      links
+        .eq(1)
+        .find(".sidebar-section-link-suffix")[0]
+        .classList.contains("unread"),
+      true,
+      "displays new messages unread indicator"
+    );
+    assert.strictEqual(
+      links
+        .eq(2)
+        .find(".sidebar-section-link-prefix svg")[0]
+        .classList.contains("d-icon-hashtag"),
+      true,
+      "displays hash icon prefix"
+    );
+    assert.strictEqual(
+      links[2].textContent.trim(),
+      "random",
+      "displays random name in a link"
+    );
+    assert.strictEqual(
+      links
+        .eq(2)
+        .find(".sidebar-section-link-suffix")[0]
+        .classList.contains("urgent"),
+      true,
+      "displays new messages mention indicator"
+    );
+  });
+
+  test("Direct messages section", async function (assert) {
+    await visit("/");
+
+    assert.strictEqual(
+      query(
+        ".sidebar-section-chat-dms .sidebar-section-header-text"
+      ).textContent.trim(),
+      "Personal chat",
```

Comment:
Let's use the `I18n` function here instead so that the test is not brittle to translation changes.